From 34e7fdc5164ff926ccffaccf50c6e970283c0142 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jan=20Jan=C3=9Fen?= <janssen@mpie.de>
Date: Sun, 4 Dec 2022 09:57:08 -0700
Subject: [PATCH] windows patch

---
 build_pynng.py |  4 ++--
 setup.py       | 11 ++++++++---
 2 files changed, 10 insertions(+), 5 deletions(-)

diff --git a/build_pynng.py b/build_pynng.py
index 93d5238..fadc76e 100644
--- a/build_pynng.py
+++ b/build_pynng.py
@@ -16,9 +16,9 @@
     # we detect ninja in the setup script; ninja and plain ol' Visual Studio put the
     # build artifacts in different places.  Kind of annoying.  Maybe it would be better
     # to modify where ninja puts its artifacts?
-    objects.append(f'./nng/build/Release/nng.lib')
+    objects.append(f'./nng/build/nng.lib')
 
-    mbedtls_dir = f'./mbedtls/build/library/Release'
+    mbedtls_dir = f'./mbedtls/build/library'
     objects += [
         mbedtls_dir + "/mbedtls.lib",
         mbedtls_dir + "/mbedx509.lib",
diff --git a/setup.py b/setup.py
index 71a09b5..9193f4a 100644
--- a/setup.py
+++ b/setup.py
@@ -17,7 +17,7 @@
 NNG_REPO = 'https://github.com/nanomsg/nng'
 NNG_REV = '4f5e11c391c4a8f1b2731aee5ad47bc0c925042a'
 MBEDTLS_REPO = 'https://github.com/ARMmbed/mbedtls.git'
-MBEDTLS_REV = '04a049bda1ceca48060b57bc4bcf5203ce591421'
+MBEDTLS_REV = '869298bffeea13b205343361b7a7daf2b210e33d'
 
 def maybe_copy(src, dst):
     os.makedirs(os.path.dirname(dst), exist_ok=True)
@@ -38,6 +38,8 @@ def build_mbedtls(cmake_args):
     cwd = f'mbedtls/build'
     os.makedirs(cwd, exist_ok=True)
     cmake_cmd = ['cmake'] + cmake_args
+    if WINDOWS:
+        cmake_cmd += ['-DCMAKE_C_COMPILER=cl', '-DCMAKE_CXX_COMPILER=cl']
     cmake_cmd += [
         '-DENABLE_PROGRAMS=OFF',
         '-DCMAKE_BUILD_TYPE=Release',
@@ -84,6 +86,8 @@ def build_nng(cmake_args):
     nng_build_dir = f'nng/build'
     os.makedirs(nng_build_dir, exist_ok=True)
     cmake_cmd = ['cmake'] + cmake_args
+    if WINDOWS:
+        cmake_cmd += ['-DCMAKE_C_COMPILER=cl', '-DCMAKE_CXX_COMPILER=cl']
     cmake_cmd += [
         '-DNNG_ENABLE_TLS=ON',
         '-DNNG_TESTS=OFF',
@@ -119,14 +123,15 @@ def build_libs():
 
     flags = ['-DCMAKE_POSITION_INDEPENDENT_CODE:BOOL=true']
     is_64bit = sys.maxsize > 2**32
-    if WINDOWS:
+    ninja_paths = shutil.which("ninja")
+    if WINDOWS and not ninja_paths:
         print('~~~building without ninja~~~', file=sys.stderr)
         if is_64bit:
             flags += ['-A', 'x64']
         else:
             flags += ['-A', 'win32']
     else:
-        if shutil.which('ninja'):
+        if ninja_paths:
             print('~~~building with ninja~~~', file=sys.stderr)
             # the ninja build generator is a million times faster.
             flags += ['-G', 'Ninja']
